problem_builder/public/js/instructor_tool.js

```diff
@@ -18,12 +18,57 @@ function InstructorToolBlock(runtime, element) {
 
         model: Result,
 
-        getCurrentPage: function(returnObject) {
-            var currentPage = this.state.currentPage;
-            if (returnObject) {
-                return this.getPage(currentPage);
+        state: {
+            order: 0
+        },
+
+        url: runtime.handlerUrl(element, 'get_result_page'),
+
+        parseState: function(response) {
+            return {
+                totalRecords: response.num_results,
+                pageSize: response.page_size
+            };
+        },
+
+        parseRecords: function(response) {
+            return _.map(response.display_data, function(row) {
+                return new Result(null, { values: row });
+            });
+        },
+
+        fetchOptions: {
+            reset: true,
+            type: 'POST',
+            contentType: 'application/json',
+            processData: false,
+            beforeSend: function(jqXHR, options) {
```

Comment:
@itsjeyd `beforeSend` is probably ok to do such type of processing in general, but there's a 3rd party instance that uses problem builders and have global `beforeSend` adding csrf and authentication data to AJAX requests. So, since ajax call arguments take precedence of global config this would cause that global `beforeSend` skipped - as a result the requests would fail due to auth or csrf protection.

I might be wrong because I observed that about half a year from now, but judging by up-to-date code it should still behave like that. So I'm suggesting rewriting this block without using `beforeSend` or making sure global `beforeSend` is called.

Let me know if you need more info - can't speak openly here as we have a NDA with those 3rd party - and it's better safe than sorry in such case :)